backend/src/govsg/services/govsg-verification-service.ts

```diff
@@ -31,20 +30,23 @@ export const sendPasscodeCreationMessage = async (
   return passcodeCreationWamid
 }
 
-const createPasscode = () => {
-  return randomInt(0, Math.pow(10, 4)).toString().padStart(4, '0')
-}
-
 export const storePrecreatedPasscode = async (
   govsgMessageId: GovsgMessage['id'],
   passcodeCreationWamid: MessageId
 ): Promise<GovsgVerification> => {
-  const passcode = createPasscode()
-  return await GovsgVerification.create({
-    govsgMessageId,
+  const govsgVerification = await GovsgVerification.findOne({
```

Comment:
Just checking on this, `upsert` did not work?

Reply:
I probably mentioned this before - upsert requires uniqueness / data from the entire row. So the work to make upsert work is significant. I feel it is not worth the effort to refactor the code by tightening constraints.

On the other hand, we could assess if we want tighter constraints in our future schema.